graph/api/src/main/scala/org/apache/spark/graph/api/CypherSession.scala

```diff
@@ -0,0 +1,285 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.graph.api
+
+import scala.collection.JavaConverters._
+
+import org.slf4j.LoggerFactory
+
+import org.apache.spark.annotation.Evolving
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.apache.spark.sql.types.{BooleanType, StructType}
+
+/**
+ * Contains constants used for convention based column naming.
+ */
+@Evolving
+object CypherSession {
+
+  /**
+   * Naming convention for identifier columns, both node and relationship identifiers.
+   */
+  val ID_COLUMN = "$ID"
+
+  /**
+   * Naming convention for relationship source identifier.
+   */
+  val SOURCE_ID_COLUMN = "$SOURCE_ID"
+
+  /**
+   * Naming convention for relationship target identifier.
+   */
+  val TARGET_ID_COLUMN = "$TARGET_ID"
+
+  /**
+   * Naming convention both for node label and relationship type prefixes.
+   */
+  val LABEL_COLUMN_PREFIX = ":"
+
+  /**
+   * Extracts [[NodeDataset]]s from a [[Dataset]] using column name conventions.
+   *
+   * For information about naming conventions, see [[CypherSession.createGraph]].
+   *
+   * @param nodes node dataset
+   * @since 3.0.0
+   */
+  def extractNodeDataset(nodes: Dataset[Row]): Set[NodeDataset] = {
+    val labelColumns = nodes.columns.filter(_.startsWith(LABEL_COLUMN_PREFIX)).toSet
+    validateLabelColumns(nodes.schema, labelColumns)
+
+    val nodeProperties = (nodes.columns.toSet - ID_COLUMN -- labelColumns)
+      .map(col => col -> col)
+      .toMap
+
+    val labelCount = labelColumns.size
+    if (labelCount > 5) {
+      LoggerFactory.getLogger(CypherSession.getClass).warn(
+        s"$labelCount label columns will result in ${Math.pow(labelCount, 2)} node frames.")
+      if (labelCount > 10) {
+        throw new IllegalArgumentException(
+          s"Expected number of label columns to be less than or equal to 10, was $labelCount.")
+      }
+    }
+
+    val labelSets = labelColumns.subsets().toSet
+
+    labelSets.map { labelSet =>
+      val predicate = labelColumns
+        .map { labelColumn =>
+          if (labelSet.contains(labelColumn)) {
+            nodes.col(labelColumn)
+          } else {
+            !nodes.col(labelColumn)
+          }
+        }
+        .reduce(_ && _)
+
+      NodeDataset(nodes.filter(predicate), ID_COLUMN, labelSet.map(_.substring(1)), nodeProperties)
+    }
+  }
+
+  /**
+   * Extracts [[RelationshipDataset]]s from a [[Dataset]] using column name conventions.
+   *
+   * For information about naming conventions, see [[CypherSession.createGraph]].
+   *
+   * @param relationships relationship dataset
+   * @since 3.0.0
+   */
+  def extractRelationshipDataset(relationships: Dataset[Row]): Set[RelationshipDataset] = {
```

Comment:
+1 for your suggestion, `extractRelationshipDataset` -> `extractRelationshipDatasets`.